File: python/problem012.py
prime_numbers = [2]
temp_values = [0]


def is_prime(n):
    for a in prime_numbers:
        if n % a == 0:
            return False
    return True


def init_temp_values():
    for b in range(0, len(temp_values)):
        temp_values[b] = 0


def get_number_of_divisors(x, y):
    init_temp_values()
    number_of_divisors = 1
    number1 = x
    number2 = y
    pointer1 = 0
    pointer2 = 0
    while number1 > 1:
        if number1 % prime_numbers[pointer1] == 0:
            number1 /= prime_numbers[pointer1]
            temp_values[pointer1] += 1
        else:
            pointer1 += 1
    while number2 > 1:
        if number2 % prime_numbers[pointer2] == 0:
            number2 /= prime_numbers[pointer2]
            temp_values[pointer2] += 1
        else:
            pointer2 += 1
    for y in temp_values:
        number_of_divisors *= y + 1
    return number_of_divisors


index = 2
check_point = 3
result = 0

while True:
    n1 = 0
    n2 = 0
    big_number = 0
    if index % 2 == 1:
        n1 = index
        n2 = (index + 1) / 2
        big_number = n1
    else:
        n1 = index / 2
        n2 = index + 1
        big_number = n2
    for i in range(check_point, big_number + 1):
        if is_prime(i):
            prime_numbers.append(i)
            temp_values.append(0)
    check_point = big_number
    index += 1
    if get_number_of_divisors(n1, n2) > 500:
        result = n1 * n2
        break

print(result)

File: javascript/problem012.js
var primeNumbers = [2];
var tempValues = [0];

function isPrime(n){
  for(a in primeNumbers){
    if(n % primeNumbers[a] == 0){
      return false;
    }
  }
  return true;
}

function initTempValues(){
  for(a in tempValues){
    tempValues[a] = 0;
  }
}

function getNumberOfDivisors(x, y){
  initTempValues();
  var numberOfDivisors = 1;
  var number1 = x;
  var number2 = y;
  var pointer1 = 0;
  var pointer2 = 0;
  while(number1 > 1){
    if(number1 % primeNumbers[pointer1] == 0){
      number1 = number1 / primeNumbers[pointer1];
      tempValues[pointer1] = tempValues[pointer1] + 1;
    }
    else{
      pointer1 = pointer1 + 1;
    }
  }
  while(number2 > 1){
    if(number2 % primeNumbers[pointer2] == 0){
      number2 = number2 / primeNumbers[pointer2];
      tempValues[pointer2] = tempValues[pointer2] + 1;
    }
    else{
      pointer2 = pointer2 + 1;
    }
  }
  for(y = 0; y < tempValues.length; y++){
      numberOfDivisors = numberOfDivisors * (tempValues[y] + 1);
  }
  return numberOfDivisors;
}

var index = 2;
var checkPoint = 3;
var result;

while(true){
  var n1;
  var n2;
  var bigNumber;
  if(index % 2 == 1){
    n1 = index;
    n2 = (index + 1) / 2;
    bigNumber = n1;
  }
  else{
    n1 = index / 2;
    n2 = index + 1
    bigNumber = n2;
  }
  for(i=checkPoint; i<=bigNumber; i++){
    if(isPrime(i)){
      primeNumbers.push(i);
      tempValues.push(0);
    }
  }
  checkPoint = bigNumber;
  index = index + 1;
  if(getNumberOfDivisors(n1, n2)  > 500){
    result = n1 * n2;
    break;
  }
}

console.log(result);
